plugins/modules/custom_image.py

```diff
@@ -178,22 +172,101 @@
 )
+from ansible.module_utils._text import to_text
+import re
+import json
+import time
+
+
+class AnsibleCloudscaleCustomImage(AnsibleCloudscaleBase):
+
+    def fetch_both_urls(self, api_call):
+        headers = self._auth_header
+
+        # Get image information
+        url = '%s%s' % (self._api_url, api_call)
+        response, info = fetch_url(self._module,
+                                   url,
+                                   headers=headers,
+                                   timeout=self._module.params['api_timeout'])
+
+        # Add import to the url
+        if api_call == 'custom-images':
+            api_call = 'custom-images/import'
+        else:
+            api_call = re.sub(r"/", "/import/", api_call)
 
+        # Get image import information
+        url = '%s%s' % (self._api_url, api_call)
+        response_import, info_import = fetch_url(self._module,
+                                                 url,
+                                                 headers=headers,
+                                                 timeout=self._module.params['api_timeout'])
 
-class AnsibleCloudscaleCustomImages(AnsibleCloudscaleBase):
+        if info_import['status'] != 200:
+            # No image is found
+            return None, info
+
+        response_import = self._module.from_json(to_text(response_import.read(), errors='surrogate_or_strict'))
+        if info['status'] != 200:
+            # There is a short period of time where no info about the image is available
+            return response_import, info_import
+
+        response = self._module.from_json(to_text(response.read(), errors='surrogate_or_strict'))
+
+        copy_keys = ['status', 'error_message', 'url']
+
+        if isinstance(response, dict):
+            # One image is found
+            if response['uuid'] == response_import['uuid']:
+                for key in copy_keys:
+                    response[key] = response_import[key]
+        else:
+            # More than one image is found
+            for r in response:
+                for i in response_import:
+                    if r['uuid'] == i['uuid']:
+                        for key in copy_keys:
+                            r[key] = i[key]
+        return response, info
+
+    def _get(self, api_call):
+        resp, info = self.fetch_both_urls(api_call)
+        if info['status'] == 200:
+            return resp
+        elif info['status'] == 404:
+            return None
+        else:
+            self._module.fail_json(msg='Failure while calling the cloudscale.ch API with GET for '
+                                       '"%s"' % api_call, fetch_url_info=info)
+
+    def _post(self, api_call, data=None):
+        ret = super()._post(api_call, data)
+        if api_call == 'custom-images/import' and ret['uuid']:
+            url = '%s%s/%s' % (self._api_url, api_call, ret['uuid'])
+            info = {}
+            info['status'] = 0
+            while info['status'] != 200:
```

Comment:
I think this should be limited to a certain amount of time. For example:

```python
for _ in range(10):
    ....
```

As it stands it would try this forever if something goes wrong, which is probably not what we want.